doc/bevindingen-fase-4.md

```diff
@@ -64,23 +76,48 @@ Ook hier geldt dat de BDI authenticatie / autorisatie aanpak in overleg tussen d
 
-Een alternatief zou kunnen zijn om de event-producerende omgeving aan te bieden als een Docker-container, zodat deze door de ontwikkelaar lokaal kan worden gedraaid.
+Deze moeilijkheden zijn inherent aan het ontwikkelen met webhooks en is door Portbase goed opgevangen door een omgeving aan te bieden waar events nagebootst kunnen worden.
 
 ## Vertalen van events
 
-In fase 3 is gekozen voor EPCIS events, terwijl in deze fase door Portbase voor DCSA events is gekozen. De DCSA events worden nu vertaald naar EPCIS om ze te kunnen propageren naar andere partijen. Er is echter een semantisch verschil tussen de typen events, wat tot problemen kan leiden: EPCIS events refereren aan de start van een gebeurtenis ("departing", "arriving", "loading" etc.), terwijl DCSA events refereren aan de afronding ervan ("departed", "loaded" etc.).  Dit kan problemen opleveren bij de interpretatie van events.
+In fase 3 is gekozen voor EPCIS events (naar aanleiding van LEO, Light Event Ontology), terwijl in deze fase door Portbase voor DCSA events is gekozen. De DCSA events worden nu in de demo vertaald naar EPCIS om ze te kunnen propageren naar andere partijen. Er is echter een semantisch verschil tussen de typen events, wat tot problemen kan leiden: EPCIS events refereren aan de start van een gebeurtenis ("departing", "arriving", "loading" etc.), terwijl DCSA events refereren aan de afronding ervan ("departed", "loaded" etc.).  Dit kan problemen opleveren bij de interpretatie van events.
+
+Voor deze use-case is DCSA een logische keuze omdat het om container verkeer gaat.  Vertalingen tussen event modellen is een vraagstuk dat nog volop in ontwikkeling is in DIL/BDI.
+
+# Geleerde lessen
+
+## Events versus notificaties
+
+Voor de events in een eerdere fase van deze demo is gekozen voor het *notificatiemodel*, waarbij alleen notificaties verstuurd worden waarna de ontvanger het daadwerkelijke event nog moet ophalen.  Dat heeft als nadeel dat de producent een endpoint moet aanbieden waar de events opgehaald kunnen worden en de ontvangende partij het event moet ophalen.
+
+In de koppeling met Portbase wordt het volledige event meteen aangeboden.  Dit heeft de implementatie aan beide zijden eenvoudiger gemaakt.
+
+## Life cycle
+
+Het is verstandig om vooraf duidelijk af te spreken hoe lang het zinvol is om geabonneerd te blijven op events van bijvoorbeeld containers en schepen.  Hierbij kunnen voorbeeld uitzonderingen doorgenomen worden zodat de ontvanger kan bepalen hoe lang deze *geïnteresseerd* blijft.
+
+## Webhooks versus event brokers
+
+Het gebruik van webhooks heeft als nadeel dat de ontvanger bereikbaar moet zijn via het gebruikte netwerk, maar is qua beheer gemakkelijker te implementeren en beheren dan een event broker.  Hoewel foutafhandeling bij het gebruik van event brokers meer voor de hand ligt, is het gemakkelijk hier van tevoren duidelijke afspraken over te maken en deze te implementeren.
```

Comment:
Bedoel je zoiets als "Hoewel foutafhandeling bij het gebruik van event brokers eenvoudiger te implementeren is" ?